src/sentry/incidents/endpoints/organization_alert_rule_index.py

```diff
@@ -37,14 +45,47 @@ def get(self, request, organization):
                 "id", flat=True
             )
 
+        teams = set(request.GET.getlist("team", []))
+        team_filter_query = None
+        if teams:
+            request_teams = request.access.teams
+            for team in request_teams:
+                if not request.access.has_team_access(team):
+                    return Response(
+                        f"Error: You do not have permission to access {team.name}",
+                        status=status.HTTP_400_BAD_REQUEST,
+                    )
```

Comment:
Anything in `request.access.teams` is already validated, we can just use it directly